CHANGELOG.md

```diff
@@ -1,5 +1,349 @@
+  for a given height, or latest.
+  ([\#1095](https://github.com/cometbft/cometbft/issues/1095))
+- `[config]` Add `[grpc.block_results_service]` gRPC configuration `BlockResultsService`
+  ([\#1095](https://github.com/cometbft/cometbft/issues/1095))
+- `[proto]` Add definitions and generated code for
+  [ADR-101](./docs/architecture/adr-101-data-companion-pull-api.md)
+  `PruningService` in the `cometbft.services.pruning.v1` proto package
+  ([\#1097](https://github.com/cometbft/cometbft/issues/1097)).
+- `[rpc/grpc]` Add privileged gRPC server and client facilities, in
+  `server/privileged` and `client/privileged` packages respectively, to
+  enable a separate API server within the node which serves trusted clients
+  without authentication and should never be exposed to public internet
+  ([\#1097](https://github.com/cometbft/cometbft/issues/1097)).
+- `[rpc/grpc]` Add a pruning service adding on the privileged gRPC server API to
+  give an [ADR-101](./docs/architecture/adr-101-data-companion-pull-api.md) data
+  companion control over block data retained by the node. The
+  `WithPruningService` option method in `server/privileged` is provided to
+  configure the pruning service
+  ([\#1097](https://github.com/cometbft/cometbft/issues/1097)).
+- `[rpc/grpc]` Add `PruningServiceClient` interface
+  for the gRPC client in `client/privileged` along with a configuration option
+  to enable it
+  ([\#1097](https://github.com/cometbft/cometbft/issues/1097)).
+- `[config]` Add `[grpc.privileged]` section to configure the privileged
+  gRPC server for the node, and `[grpc.privileged.pruning_service]` section
+  to control the pruning service
+  ([\#1097](https://github.com/cometbft/cometbft/issues/1097)).
+- `[metrics]` Add metrics to monitor pruning and current available data in stores: `PruningServiceBlockRetainHeight`, `PruningServiceBlockResultsRetainHeight`, `ApplicationBlockRetainHeight`, `BlockStoreBaseHeight`, `ABCIResultsBaseHeight`. 
+  ([\#1234](https://github.com/cometbft/cometbft/pull/1234))
+`[rpc/grpc]` Add gRPC endpoint for pruning the block and transaction indexes
+([\#1327](https://github.com/cometbft/cometbft/pull/1327))
+- `[state]` Add TxIndexer and BlockIndexer pruning metrics
+  ([\#1334](https://github.com/cometbft/cometbft/issues/1334))
+- `[metrics]` Add metric for mempool size in bytes `SizeBytes`.
+  ([\#1512](https://github.com/cometbft/cometbft/pull/1512))
+- `[mempool]` Add `nop` mempool ([\#1643](https://github.com/cometbft/cometbft/pull/1643))
+
+  If you want to use it, change mempool's `type` to `nop`:
+
+  ```toml
+  [mempool]
+
+  # The type of mempool for this node to use.
+  #
+  # Possible types:
+  # - "flood" : concurrent linked list mempool with flooding gossip protocol
+  # (default)
+  # - "nop"   : nop-mempool (short for no operation; the ABCI app is responsible
+  # for storing, disseminating and proposing txs). "create_empty_blocks=false"
+  # is not supported.
```

Comment:
The comments lifted from the self-documenting example are perhaps too bulky for the changelog.
```suggestion
```

Perhaps the "nop" mempool should be mentioned in the summary as well?

Reply:
I personally think it's fine to include this kind of info in the changelog. We should actually be doing more of this kind of thing, since changelog entries need to be user impact-focused as opposed to just being slightly re-worded Git commit messages.

And I can add something about the `nop` mempool in the summary :+1: